src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -914,7 +931,7 @@ public bool CleanCacheAndDisposeIfUnused()
                     // if a pool was used since the last time we cleaned up, give it another chance. New pools
                     // start out saying they've recently been used, to give them a bit of breathing room and time
                     // for the initial collection to be added to it.
-                    if (_associatedConnectionCount == 0 && !_usedSinceLastCleanup)
+                    if (_associatedConnectionCount == 0 && !_usedSinceLastCleanup && _http2Connection == null)
```

Comment:
I don't think we are updating _usedSinceLastCleanup in the HTTP2 case.

Reply:
I add update to it when we grab existing connection. 
I need to do more testing but I think it may be enough.